engine/commands/cortex_upd_cmd.h

```diff
@@ -1,13 +1,124 @@
 #pragma once
-#include <string>
 #include <optional>
+#include <string>
+
+#include "httplib.h"
+#include "nlohmann/json.hpp"
+#include "utils/file_manager_utils.h"
+#include "utils/logging_utils.h"
 
 namespace commands {
+#ifndef CORTEX_VARIANT
+#define CORTEX_VARIANT file_manager_utils::kProdVariant
+#endif
+constexpr const auto kNightlyHost = "https://delta.jan.ai";
+constexpr const auto kNightlyFileName = "cortex-nightly.tar.gz";
+const std::string kCortexBinary = "cortex";
+
+inline std::string GetCortexBinary() {
+#if defined(_WIN32)
+  constexpr const bool has_exe = true;
+#else
+  constexpr const bool has_exe = false;
+#endif
+  if (CORTEX_VARIANT == file_manager_utils::kNightlyVariant) {
+    return has_exe ? kCortexBinary + "-nightly.exe"
+                   : kCortexBinary + "-nightly";
+  } else if (CORTEX_VARIANT == file_manager_utils::kBetaVariant) {
+    return has_exe ? kCortexBinary + "-beta.exe" : kCortexBinary + "-beta";
+  } else {
+    return has_exe ? kCortexBinary + ".exe" : kCortexBinary;
+  }
+}
+
+inline std::string GetHostName() {
+  if (CORTEX_VARIANT == file_manager_utils::kNightlyVariant) {
```

Comment:
Will this become a switch when we have `beta`, `nightly`?

Reply:
C++ does not support switch for string. We only use 1 function to get `stable` and `beta` build.